internal/util/grpcclient/client.go

```diff
@@ -249,7 +249,15 @@ func (c *ClientBase[T]) callOnce(ctx context.Context, caller func(client T) (any
 		return generic.Zero[T](), err
 	}
 
-	ret, err2 := caller(client)
+	var (
+		ret  any
+		err2 error
+	)
+
+	_, _ = retry.DoGrpc(uint(c.MaxAttempts), func() (any, error) {
```

Comment:
do we still need recall in grpc once we have retry?

Reply:
yes, because the `recall` will execute the second `call` only if the first execution `call` gets an error. If the `call` gets an error, there is an error in grpc.